aten/src/THNN/generic/VolumetricDilatedMaxPooling.c

```diff
@@ -381,16 +372,13 @@ static void THNN_(VolumetricDilatedMaxPooling_updateGradInput_frame)(
         for (j = 0; j < owidth; j++)
         {
           /* retrieve position of max */
-          THIndex_t * indzp = &indz_p_k[ti * oheight * owidth + i * owidth + j];
-          int64_t maxti = ((unsigned char*)(indzp))[0] * dilationT + ti * dT - pT;
-          int64_t maxi  = ((unsigned char*)(indzp))[1] * dilationH + i * dH - pH;
-          int64_t maxj  = ((unsigned char*)(indzp))[2] * dilationW + j * dW - pW;
-
-	  if (maxti != -1) {
-	    /* update gradient */
-	    gradInput_p_k[maxti * iheight * iwidth + maxi * iwidth + maxj] +=
-	      gradOutput_p_k[ti * oheight * owidth + i * owidth + j];
-	  }
+          int64_t index = ti * oheight * owidth + i * owidth + j;
+          int64_t maxp = indz_p_k[index] - TH_INDEX_BASE;
+
+          if (maxp != -1) {
+            /* update gradient */
+            gradInput_p_k[maxp] += gradOutput_p_k[index];
```

Comment:
Is it sufficient to just assign gradInput_p_k[maxp] = gradOutput_p_k[index]?

Reply:
Nah, different outputs can depend on the same input.